src/pages/patientView/mutation/MSKMutationTable.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import SampleManager from "../sampleManager";
+import PatientHeader from "../patientHeader/PatientHeader";
+import {PatientViewPageStore} from "../clinicalInformation/PatientViewPageStore";
+import {default as MSKTable, Column} from "../../../shared/components/msktable/MSKTable";
+import {Mutation} from "../../../shared/api/generated/CBioPortalAPI";
+import DiscreteCNAColumnFormatter from "./column/DiscreteCNAColumnFormatter";
+import {MutSigData} from "../PatientViewPage";
+import MrnaExprColumnFormatter from "./column/MrnaExprColumnFormatter";
+import CohortColumnFormatter from "./column/CohortColumnFormatter";
+
+export type MSKMutationTableProps = {
+    sampleManager:SampleManager | null;
+    store:PatientViewPageStore;
+    mutSigData?:MutSigData;
+}
+
+class MutationTableComponent extends MSKTable<Mutation[]> {
+}
+
+@observer
+export default class MSKMutationTable extends React.Component<MSKMutationTableProps,{}> {
+    render() {
+        const columns:Column<Mutation[]>[] = [];
```

Comment:
I changed it to a compromise - the columns are stored in an object but the user passes in an array of columns which encodes both visibility and order. The management of visibility and order as an object can occur at a higher level, and be passed in as an array. I think this is cleaner.